packages/insomnia-app/app/ui/components/base/file-input-button.js

```diff
@@ -65,15 +66,15 @@ class FileInputButton extends React.PureComponent<Props> {
       options.filters = [{ name: 'Files', extensions: this.props.extensions }];
     }
 
-    remote.dialog.showOpenDialog(options, async paths => {
-      // Only change the file if a new file was selected
-      if (!paths || paths.length === 0) {
-        return;
-      }
+    const { canceled, filePaths } = await remote.dialog.showOpenDialog(options);
+
+    // Only change the file if a new file was selected
+    if (canceled && filePaths.length) {
```

Comment:
Should this be the following?

```suggestion
    if (canceled || !filePaths.length) {
```

The two are not the same 🤔

Reply:
The Electron docs say the array will be empty if `canceled` is `true` so I don't think there's a need to check the length. I'll just remove it

Reply:
Also, good catch! I definitely tested that change after making it but must have done a breaking refactor right afterwards without realizing.